diff -c -r grammars/semantic-c-by.el new/semantic-c-by.el
*** semantic-c-by.el	2011-07-31 20:19:31.884753047 -0400
--- c-by.el	2011-07-31 20:21:02.694753113 -0400
***************
*** 1,42 ****
! ;;; semantic-c-by.el --- Generated parser support file
  
! ;; Copyright (C) 1999-2011 Free Software Foundation, Inc.
  
! ;; Author: Chong Yidong <cyd@furball>
! ;; Created: 2011-07-31 20:19:31-0400
! ;; Keywords: syntax
! ;; X-RCS: $Id$
  
! ;; This file is not part of GNU Emacs.
  
! ;; This program is free software; you can redistribute it and/or
! ;; modify it under the terms of the GNU General Public License as
! ;; published by the Free Software Foundation, either version 3 of
! ;; the License, or (at your option) any later version.
! 
! ;; This software is distributed in the hope that it will be useful,
  ;; but WITHOUT ANY WARRANTY; without even the implied warranty of
! ;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
! ;; General Public License for more details.
! ;;
  ;; You should have received a copy of the GNU General Public License
! ;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
  
  ;;; Commentary:
  ;;
! ;; PLEASE DO NOT MANUALLY EDIT THIS FILE!  It is automatically
! ;; generated from the grammar file c.by.
! 
! ;;; History:
! ;;
  
  ;;; Code:
! 
! ;;; Prologue
! ;;
! 
! ;;; Declarations
! ;;
  (defconst semantic-c-by--keyword-table
    (semantic-lex-make-keyword-table
     '(("extern" . EXTERN)
--- 1,36 ----
! ;;; semantic/bovine/c-by.el --- Generated parser support file
  
! ;;; Copyright (C) 1999, 2000, 2001, 2002, 2003, 2004, 2005, 2006,
! ;;; 2007, 2008, 2009, 2010 Free Software Foundation, Inc.
  
! ;; This file is part of GNU Emacs.
  
! ;; GNU Emacs is free software: you can redistribute it and/or modify
! ;; it under the terms of the GNU General Public License as published by
! ;; the Free Software Foundation, either version 3 of the License, or
! ;; (at your option) any later version.
  
! ;; GNU Emacs is distributed in the hope that it will be useful,
  ;; but WITHOUT ANY WARRANTY; without even the implied warranty of
! ;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
! ;; GNU General Public License for more details.
! 
  ;; You should have received a copy of the GNU General Public License
! ;; along with GNU Emacs.  If not, see <http://www.gnu.org/licenses/>.
  
  ;;; Commentary:
  ;;
! ;; This file was generated from etc/grammars/c.by.
  
  ;;; Code:
! 
! (require 'semantic/lex)
! (eval-when-compile (require 'semantic/bovine))
! 
! (declare-function semantic-c-reconstitute-token "semantic/bovine/c")
! (declare-function semantic-c-reconstitute-template "semantic/bovine/c")
! (declare-function semantic-expand-c-tag "semantic/bovine/c")
! 
  (defconst semantic-c-by--keyword-table
    (semantic-lex-make-keyword-table
     '(("extern" . EXTERN)
***************
*** 189,199 ****
  
  (defconst semantic-c-by--parse-table
    `(
!     (bovine-toplevel 
       (declaration)
       ) ;; end bovine-toplevel
  
!     (bovine-inner-scope 
       (codeblock)
       ) ;; end bovine-inner-scope
  
--- 183,193 ----
  
  (defconst semantic-c-by--parse-table
    `(
!     (bovine-toplevel
       (declaration)
       ) ;; end bovine-toplevel
  
!     (bovine-inner-scope
       (codeblock)
       ) ;; end bovine-inner-scope
  
***************
*** 1245,1253 ****
  	 (nth 7 vals))
  	(nth 0 vals)
  	(nth 10 vals)
! 	(list
! 	 (nth 4 vals))
! 	(nth 9 vals))
        )
       (opt-stars
        opt-class
--- 1239,1245 ----
  	 (nth 7 vals))
  	(nth 0 vals)
  	(nth 10 vals)
! 	(nth 4 vals))
        )
       (opt-stars
        opt-class
***************
*** 1269,1277 ****
  	 (nth 6 vals))
  	(nth 0 vals)
  	(nth 9 vals)
! 	(list
! 	 (nth 4 vals))
! 	(nth 8 vals))
        )
       ) ;; end func-decl
  
--- 1261,1267 ----
  	 (nth 6 vals))
  	(nth 0 vals)
  	(nth 9 vals)
! 	(nth 4 vals))
        )
       ) ;; end func-decl
  
***************
*** 1442,1452 ****
        namespace-symbol
        opt-bits
        opt-array
        ,(semantic-lambda
  	(nth 2 vals)
  	(nth 0 vals)
  	(nth 3 vals)
! 	(nth 4 vals))
        )
       ) ;; end varname
  
--- 1432,1444 ----
        namespace-symbol
        opt-bits
        opt-array
+       opt-assign
        ,(semantic-lambda
  	(nth 2 vals)
  	(nth 0 vals)
  	(nth 3 vals)
! 	(nth 4 vals)
! 	(nth 5 vals))
        )
       ) ;; end varname
  
***************
*** 1491,1518 ****
        )
       ) ;; end variablearg-opt-name
  
-     (varname-opt-initializer
-      (semantic-list)
-      (opt-assign)
-      ( ;;EMPTY
-       )
-      ) ;; end varname-opt-initializer
- 
      (varnamelist
       (opt-ref
        varname
-       varname-opt-initializer
        punctuation
        "\\`[,]\\'"
        varnamelist
        ,(semantic-lambda
  	(cons
  	 (nth 1 vals)
! 	 (nth 4 vals)))
        )
       (opt-ref
        varname
-       varname-opt-initializer
        ,(semantic-lambda
  	(list
  	 (nth 1 vals)))
--- 1483,1501 ----
        )
       ) ;; end variablearg-opt-name
  
      (varnamelist
       (opt-ref
        varname
        punctuation
        "\\`[,]\\'"
        varnamelist
        ,(semantic-lambda
  	(cons
  	 (nth 1 vals)
! 	 (nth 3 vals)))
        )
       (opt-ref
        varname
        ,(semantic-lambda
  	(list
  	 (nth 1 vals)))
***************
*** 2124,2187 ****
        "\\`[&]\\'")
       ) ;; end expr-start
  
-     (expr-binop
-      (punctuation
-       "\\`[-]\\'")
-      (punctuation
-       "\\`[+]\\'")
-      (punctuation
-       "\\`[*]\\'")
-      (punctuation
-       "\\`[/]\\'")
-      (punctuation
-       "\\`[&]\\'"
-       punctuation
-       "\\`[&]\\'")
-      (punctuation
-       "\\`[&]\\'")
-      (punctuation
-       "\\`[|]\\'"
-       punctuation
-       "\\`[|]\\'")
-      (punctuation
-       "\\`[|]\\'")
-      ) ;; end expr-binop
- 
      (expression
!      (unaryexpression
!       expr-binop
!       unaryexpression
        ,(semantic-lambda
  	(list
  	 (identity start)
  	 (identity end)))
        )
!      (unaryexpression
        ,(semantic-lambda
  	(list
  	 (identity start)
  	 (identity end)))
        )
-      ) ;; end expression
- 
-     (unaryexpression
-      (number)
-      (multi-stage-dereference)
       (NEW
!       multi-stage-dereference)
       (NEW
        builtintype-types
!       semantic-list)
!      (namespace-symbol)
!      (string-seq)
       (type-cast
!       expression)
       (semantic-list
!       expression)
!      (semantic-list)
       (expr-start
!       expression)
!      ) ;; end unaryexpression
      )
    "Parser table.")
  
--- 2107,2180 ----
        "\\`[&]\\'")
       ) ;; end expr-start
  
      (expression
!      (number
        ,(semantic-lambda
  	(list
  	 (identity start)
  	 (identity end)))
        )
!      (multi-stage-dereference
        ,(semantic-lambda
  	(list
  	 (identity start)
  	 (identity end)))
        )
       (NEW
!       multi-stage-dereference
!       ,(semantic-lambda
! 	(list
! 	 (identity start)
! 	 (identity end)))
!       )
       (NEW
        builtintype-types
!       semantic-list
!       ,(semantic-lambda
! 	(list
! 	 (identity start)
! 	 (identity end)))
!       )
!      (namespace-symbol
!       ,(semantic-lambda
! 	(list
! 	 (identity start)
! 	 (identity end)))
!       )
!      (string-seq
!       ,(semantic-lambda
! 	(list
! 	 (identity start)
! 	 (identity end)))
!       )
       (type-cast
!       expression
!       ,(semantic-lambda
! 	(list
! 	 (identity start)
! 	 (identity end)))
!       )
!      (semantic-list
!       expression
!       ,(semantic-lambda
! 	(list
! 	 (identity start)
! 	 (identity end)))
!       )
       (semantic-list
!       ,(semantic-lambda
! 	(list
! 	 (identity start)
! 	 (identity end)))
!       )
       (expr-start
!       expression
!       ,(semantic-lambda
! 	(list
! 	 (identity start)
! 	 (identity end)))
!       )
!      ) ;; end expression
      )
    "Parser table.")
  
***************
*** 2193,2208 ****
  	semantic-flex-keywords-obarray semantic-c-by--keyword-table
  	semantic-equivalent-major-modes '(c-mode c++-mode)
  	))
- 
- 
- ;;; Analyzers
- ;;
- (require 'semantic-lex)
- 
  
  ;;; Epilogue
  ;;
  
! (provide 'semantic-c-by)
  
! ;;; semantic-c-by.el ends here
--- 2186,2196 ----
  	semantic-flex-keywords-obarray semantic-c-by--keyword-table
  	semantic-equivalent-major-modes '(c-mode c++-mode)
  	))
  
  ;;; Epilogue
  ;;
  
! (provide 'semantic/bovine/c-by)
  
! ;; arch-tag: 27da9f71-d2ef-473f-92a7-b0006b1a8491
! ;;; semantic/bovine/c-by.el ends here
diff -c -r grammars/semantic-make-by.el new/semantic-make-by.el
*** semantic-make-by.el	2011-07-31 20:19:32.304753048 -0400
--- make-by.el	2011-07-31 20:21:02.694753113 -0400
***************
*** 1,36 ****
! ;;; semantic-make-by.el --- Generated parser support file
  
! ;; Copyright (C) 1999-2011 Free Software Foundation, Inc.
  
! ;; Author: Chong Yidong <cyd@furball>
! ;; Created: 2011-07-31 20:19:32-0400
! ;; Keywords: syntax
! ;; X-RCS: $Id$
  
! ;; This file is not part of GNU Emacs.
  
! ;; This program is free software; you can redistribute it and/or
! ;; modify it under the terms of the GNU General Public License as
! ;; published by the Free Software Foundation, either version 3 of
! ;; the License, or (at your option) any later version.
! 
! ;; This software is distributed in the hope that it will be useful,
  ;; but WITHOUT ANY WARRANTY; without even the implied warranty of
! ;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
! ;; General Public License for more details.
! ;;
  ;; You should have received a copy of the GNU General Public License
! ;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
  
  ;;; Commentary:
  ;;
! ;; PLEASE DO NOT MANUALLY EDIT THIS FILE!  It is automatically
! ;; generated from the grammar file make.by.
! 
! ;;; History:
! ;;
  
  ;;; Code:
  
  ;;; Prologue
  ;;
--- 1,32 ----
! ;;; semantic/bovine/make-by.el --- Generated parser support file
  
! ;; Copyright (C) 1999, 2000, 2001, 2002, 2003, 2004, 2008, 2009, 2010
! ;;   Free Software Foundation, Inc.
  
! ;; This file is part of GNU Emacs.
  
! ;; GNU Emacs is free software: you can redistribute it and/or modify
! ;; it under the terms of the GNU General Public License as published by
! ;; the Free Software Foundation, either version 3 of the License, or
! ;; (at your option) any later version.
  
! ;; GNU Emacs is distributed in the hope that it will be useful,
  ;; but WITHOUT ANY WARRANTY; without even the implied warranty of
! ;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
! ;; GNU General Public License for more details.
! 
  ;; You should have received a copy of the GNU General Public License
! ;; along with GNU Emacs.  If not, see <http://www.gnu.org/licenses/>.
  
  ;;; Commentary:
  ;;
! ;; This file was generated from etc/grammars/make.by.
  
  ;;; Code:
+ 
+ (require 'semantic/lex)
+ (eval-when-compile (require 'semantic/bovine))
+ 
  
  ;;; Prologue
  ;;
***************
*** 70,76 ****
  
  (defconst semantic-make-by--parse-table
    `(
!     (bovine-toplevel 
       (Makefile)
       ) ;; end bovine-toplevel
  
--- 66,72 ----
  
  (defconst semantic-make-by--parse-table
    `(
!     (bovine-toplevel
       (Makefile)
       ) ;; end bovine-toplevel
  
***************
*** 385,399 ****
  	semantic-flex-keywords-obarray semantic-make-by--keyword-table
  	))
  
! 
! ;;; Analyzers
! ;;
! (require 'semantic-lex)
! 
! 
! ;;; Epilogue
! ;;
! 
! (provide 'semantic-make-by)
  
! ;;; semantic-make-by.el ends here
--- 381,387 ----
  	semantic-flex-keywords-obarray semantic-make-by--keyword-table
  	))
  
! (provide 'semantic/bovine/make-by)
  
! ;; arch-tag: 8c3749b8-7b19-46e0-af01-2a4d02fd0352
! ;;; semantic/bovine/make-by.el ends here
diff -c -r grammars/semantic-scm-by.el new/semantic-scm-by.el
*** semantic-scm-by.el	2011-07-31 20:19:32.434753048 -0400
--- scm-by.el	2011-07-31 20:21:02.694753113 -0400
***************
*** 1,36 ****
  ;;; semantic-scm-by.el --- Generated parser support file
  
! ;; Copyright (C) 2001-2011 Free Software Foundation, Inc.
  
! ;; Author: Chong Yidong <cyd@furball>
! ;; Created: 2011-07-31 20:19:32-0400
! ;; Keywords: syntax
! ;; X-RCS: $Id$
  
! ;; This file is not part of GNU Emacs.
  
! ;; This program is free software; you can redistribute it and/or
! ;; modify it under the terms of the GNU General Public License as
! ;; published by the Free Software Foundation, either version 3 of
! ;; the License, or (at your option) any later version.
! 
! ;; This software is distributed in the hope that it will be useful,
  ;; but WITHOUT ANY WARRANTY; without even the implied warranty of
! ;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
! ;; General Public License for more details.
! ;;
  ;; You should have received a copy of the GNU General Public License
! ;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
  
  ;;; Commentary:
  ;;
! ;; PLEASE DO NOT MANUALLY EDIT THIS FILE!  It is automatically
! ;; generated from the grammar file scheme.by.
! 
! ;;; History:
! ;;
  
  ;;; Code:
  
  ;;; Prologue
  ;;
--- 1,31 ----
  ;;; semantic-scm-by.el --- Generated parser support file
  
! ;; Copyright (C) 2001, 2003, 2009, 2010 Free Software Foundation, Inc.
  
! ;; This file is part of GNU Emacs.
  
! ;; GNU Emacs is free software: you can redistribute it and/or modify
! ;; it under the terms of the GNU General Public License as published by
! ;; the Free Software Foundation, either version 3 of the License, or
! ;; (at your option) any later version.
  
! ;; GNU Emacs is distributed in the hope that it will be useful,
  ;; but WITHOUT ANY WARRANTY; without even the implied warranty of
! ;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
! ;; GNU General Public License for more details.
! 
  ;; You should have received a copy of the GNU General Public License
! ;; along with GNU Emacs.  If not, see <http://www.gnu.org/licenses/>.
  
  ;;; Commentary:
  ;;
! ;; This file was generated from etc/grammars/scm.by.
  
  ;;; Code:
+ 
+ (require 'semantic/lex)
+ 
+ (eval-when-compile (require 'semantic/bovine))
  
  ;;; Prologue
  ;;
***************
*** 58,64 ****
  
  (defconst semantic-scm-by--parse-table
    `(
!     (bovine-toplevel 
       (scheme)
       ) ;; end bovine-toplevel
  
--- 53,59 ----
  
  (defconst semantic-scm-by--parse-table
    `(
!     (bovine-toplevel
       (scheme)
       ) ;; end bovine-toplevel
  
***************
*** 190,204 ****
  	semantic-flex-keywords-obarray semantic-scm-by--keyword-table
  	))
  
! 
! ;;; Analyzers
! ;;
! (require 'semantic-lex)
! 
! 
! ;;; Epilogue
! ;;
! 
! (provide 'semantic-scm-by)
  
! ;;; semantic-scm-by.el ends here
--- 185,191 ----
  	semantic-flex-keywords-obarray semantic-scm-by--keyword-table
  	))
  
! (provide 'semantic/bovine/scm-by)
  
! ;; arch-tag: 945749cc-a039-434f-bf95-bf2480a506c3
! ;;; semantic/bovine/scm-by.el ends here
diff -c -r grammars/wisent-javascript-jv-wy.el new/wisent-javascript-jv-wy.el
*** wisent-javascript-jv-wy.el	2011-07-31 20:19:49.284753061 -0400
--- js-wy.el	2011-07-31 20:21:11.574753119 -0400
***************
*** 1,36 ****
! ;;; wisent-javascript-jv-wy.el --- Generated parser support file
  
! ;; Copyright (C) 2005-2011 Free Software Foundation, Inc.
  
! ;; Author: Chong Yidong <cyd@furball>
! ;; Created: 2011-07-31 20:19:49-0400
! ;; Keywords: syntax
! ;; X-RCS: $Id$
  
! ;; This file is not part of GNU Emacs.
  
! ;; This program is free software; you can redistribute it and/or
! ;; modify it under the terms of the GNU General Public License as
! ;; published by the Free Software Foundation, either version 3 of
! ;; the License, or (at your option) any later version.
  
! ;; This software is distributed in the hope that it will be useful,
  ;; but WITHOUT ANY WARRANTY; without even the implied warranty of
! ;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
! ;; General Public License for more details.
! ;;
  ;; You should have received a copy of the GNU General Public License
! ;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
  
  ;;; Commentary:
  ;;
! ;; PLEASE DO NOT MANUALLY EDIT THIS FILE!  It is automatically
! ;; generated from the grammar file js.wy.
! 
! ;;; History:
  ;;
  
  ;;; Code:
  
  ;;; Prologue
  ;;
--- 1,66 ----
! ;;; semantic/wisent/js-wy.el --- Generated parser support file
  
! ;; Copyright (C) 2005, 2009, 2010  Free Software Foundation, Inc.
! ;; Copyright (C) 1998-2011 Ecma International.
  
! ;; Author: Joakim Verona
  
! ;; This file is part of GNU Emacs.
  
! ;; GNU Emacs is free software: you can redistribute it and/or modify
! ;; it under the terms of the GNU General Public License as published by
! ;; the Free Software Foundation, either version 3 of the License, or
! ;; (at your option) any later version.
  
! ;; GNU Emacs is distributed in the hope that it will be useful,
  ;; but WITHOUT ANY WARRANTY; without even the implied warranty of
! ;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
! ;; GNU General Public License for more details.
! 
  ;; You should have received a copy of the GNU General Public License
! ;; along with GNU Emacs.  If not, see <http://www.gnu.org/licenses/>.
  
  ;;; Commentary:
  ;;
! ;; This file was generated from etc/grammars/js.wy.
! ;;
! ;; It is derived from the grammar in the ECMAScript Language
! ;; Specification published at
! ;;
! ;; http://www.ecma-international.org/publications/standards/Ecma-262.htm
  ;;
+ ;; and redistributed under the following license:
+ ;;
+ ;; Redistribution and use in source and binary forms, with or without
+ ;; modification, are permitted provided that the following conditions
+ ;; are met:
+ ;;
+ ;; 1. Redistributions of source code must retain the above copyright
+ ;; notice, this list of conditions and the following disclaimer.
+ ;;
+ ;; 2. Redistributions in binary form must reproduce the above
+ ;; copyright notice, this list of conditions and the following
+ ;; disclaimer in the documentation and/or other materials provided
+ ;; with the distribution.
+ ;;
+ ;; 3. Neither the name of the authors nor Ecma International may be
+ ;; used to endorse or promote products derived from this software
+ ;; without specific prior written permission.  THIS SOFTWARE IS
+ ;; PROVIDED BY THE ECMA INTERNATIONAL "AS IS" AND ANY EXPRESS OR
+ ;; IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ ;; WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ ;; ARE DISCLAIMED. IN NO EVENT SHALL ECMA INTERNATIONAL BE LIABLE FOR
+ ;; ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ ;; CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+ ;; OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+ ;; BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ ;; LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ ;; (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+ ;; USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ ;; DAMAGE.
  
  ;;; Code:
+ (require 'semantic/lex)
  
  ;;; Prologue
  ;;
***************
*** 377,383 ****
     '((parse-stream . wisent-parse-stream)))
    (setq semantic-parser-name "LALR"
  	semantic--parse-table wisent-javascript-jv-wy--parse-table
! 	semantic-debug-parser-source "js.wy"
  	semantic-flex-keywords-obarray wisent-javascript-jv-wy--keyword-table
  	semantic-lex-types-obarray wisent-javascript-jv-wy--token-table)
    ;; Collect unmatched syntax lexical tokens
--- 407,413 ----
     '((parse-stream . wisent-parse-stream)))
    (setq semantic-parser-name "LALR"
  	semantic--parse-table wisent-javascript-jv-wy--parse-table
! 	semantic-debug-parser-source "wisent-javascript-jv.wy"
  	semantic-flex-keywords-obarray wisent-javascript-jv-wy--keyword-table
  	semantic-lex-types-obarray wisent-javascript-jv-wy--token-table)
    ;; Collect unmatched syntax lexical tokens
***************
*** 388,395 ****
  
  ;;; Analyzers
  ;;
- (require 'semantic-lex)
- 
  (define-lex-keyword-type-analyzer wisent-javascript-jv-wy--<keyword>-keyword-analyzer
    "keyword analyzer for <keyword> tokens."
    "\\(\\sw\\|\\s_\\)+")
--- 418,423 ----
***************
*** 495,500 ****
    semantic-lex-default-action
    )
  
! (provide 'wisent-javascript-jv-wy)
  
! ;;; wisent-javascript-jv-wy.el ends here
--- 523,529 ----
    semantic-lex-default-action
    )
  
! (provide 'semantic/wisent/js-wy)
  
! ;; arch-tag: ff9ecf05-18e6-46a6-a3bb-e7f43f04640f
! ;;; semantic/wisent/js-wy.el ends here
diff -c -r grammars/wisent-java-tags-wy.el new/wisent-java-tags-wy.el
*** wisent-java-tags-wy.el	2011-07-31 20:19:48.674753059 -0400
--- javat-wy.el	2011-07-31 20:21:11.574753119 -0400
***************
*** 1,36 ****
! ;;; wisent-java-tags-wy.el --- Generated parser support file
  
! ;; Copyright (C) 2002-2011 Free Software Foundation, Inc.
  
! ;; Author: Chong Yidong <cyd@furball>
! ;; Created: 2011-07-31 20:19:48-0400
! ;; Keywords: syntax
! ;; X-RCS: $Id$
  
! ;; This file is not part of GNU Emacs.
  
! ;; This program is free software; you can redistribute it and/or
! ;; modify it under the terms of the GNU General Public License as
! ;; published by the Free Software Foundation, either version 3 of
! ;; the License, or (at your option) any later version.
! 
! ;; This software is distributed in the hope that it will be useful,
  ;; but WITHOUT ANY WARRANTY; without even the implied warranty of
! ;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
! ;; General Public License for more details.
! ;;
  ;; You should have received a copy of the GNU General Public License
! ;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
  
  ;;; Commentary:
  ;;
! ;; PLEASE DO NOT MANUALLY EDIT THIS FILE!  It is automatically
! ;; generated from the grammar file java-tags.wy.
! 
! ;;; History:
! ;;
  
  ;;; Code:
  
  ;;; Prologue
  ;;
--- 1,31 ----
! ;;; semantic/wisent/javat-wy.el --- Generated parser support file
  
! ;; Copyright (C) 2002, 2007, 2009, 2010  Free Software Foundation, Inc.
  
! ;; This file is part of GNU Emacs.
  
! ;; GNU Emacs is free software: you can redistribute it and/or modify
! ;; it under the terms of the GNU General Public License as published by
! ;; the Free Software Foundation, either version 3 of the License, or
! ;; (at your option) any later version.
  
! ;; GNU Emacs is distributed in the hope that it will be useful,
  ;; but WITHOUT ANY WARRANTY; without even the implied warranty of
! ;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
! ;; GNU General Public License for more details.
! 
  ;; You should have received a copy of the GNU General Public License
! ;; along with GNU Emacs.  If not, see <http://www.gnu.org/licenses/>.
  
  ;;; Commentary:
  ;;
! ;; This file was generated from the grammar file
! ;; semantic/wisent/wisent-java-tags.wy in the CEDET repository.
  
  ;;; Code:
+ 
+ (require 'semantic/lex)
+ 
  
  ;;; Prologue
  ;;
***************
*** 181,187 ****
       ("continue" summary "continue [<label>] ;")
       ("const" summary "Unused reserved word")
       ("class" summary "Class declaration: class <name>")
!      ("char" summary "Integral primitive type (' ' to '￿') (0 to 65535)")
       ("catch" summary "try {<stmts>} catch(<parm>) {<stmts>} ... ")
       ("case" summary "switch(<expr>) {case <const-expr>: <stmts> ... }")
       ("byte" summary "Integral primitive type (-128 to 127)")
--- 176,182 ----
       ("continue" summary "continue [<label>] ;")
       ("const" summary "Unused reserved word")
       ("class" summary "Class declaration: class <name>")
!      ("char" summary "Integral primitive type ('\000' to '￿') (0 to 65535)")
       ("catch" summary "try {<stmts>} catch(<parm>) {<stmts>} ... ")
       ("case" summary "switch(<expr>) {case <const-expr>: <stmts> ... }")
       ("byte" summary "Integral primitive type (-128 to 127)")
***************
*** 402,408 ****
  	((SEMICOLON))
  	((block)))
         (block
! 	((BRACE_BLOCK)))
         (formal_parameter_list
  	((PAREN_BLOCK)
  	 (semantic-parse-region
--- 397,403 ----
  	((SEMICOLON))
  	((block)))
         (block
! 	   ((BRACE_BLOCK)))
         (formal_parameter_list
  	((PAREN_BLOCK)
  	 (semantic-parse-region
***************
*** 563,569 ****
     '((parse-stream . wisent-parse-stream)))
    (setq semantic-parser-name "LALR"
  	semantic--parse-table wisent-java-tags-wy--parse-table
! 	semantic-debug-parser-source "java-tags.wy"
  	semantic-flex-keywords-obarray wisent-java-tags-wy--keyword-table
  	semantic-lex-types-obarray wisent-java-tags-wy--token-table)
    ;; Collect unmatched syntax lexical tokens
--- 558,564 ----
     '((parse-stream . wisent-parse-stream)))
    (setq semantic-parser-name "LALR"
  	semantic--parse-table wisent-java-tags-wy--parse-table
! 	semantic-debug-parser-source "wisent-java-tags.wy"
  	semantic-flex-keywords-obarray wisent-java-tags-wy--keyword-table
  	semantic-lex-types-obarray wisent-java-tags-wy--token-table)
    ;; Collect unmatched syntax lexical tokens
***************
*** 574,581 ****
  
  ;;; Analyzers
  ;;
- (require 'semantic-lex)
- 
  (define-lex-keyword-type-analyzer wisent-java-tags-wy--<keyword>-keyword-analyzer
    "keyword analyzer for <keyword> tokens."
    "\\(\\sw\\|\\s_\\)+")
--- 569,574 ----
***************
*** 684,689 ****
    ;;;;
    semantic-lex-default-action)
  
! (provide 'wisent-java-tags-wy)
  
! ;;; wisent-java-tags-wy.el ends here
--- 677,683 ----
    ;;;;
    semantic-lex-default-action)
  
! (provide 'semantic/wisent/javat-wy)
  
! ;; arch-tag: b035ba00-e014-4c71-9bc7-a9dbf3d5b911
! ;;; semantic/wisent/javat-wy.el ends here
diff -c -r grammars/wisent-python-wy.el new/wisent-python-wy.el
*** wisent-python-wy.el	2011-07-31 20:19:50.254753061 -0400
--- python-wy.el	2011-07-31 20:21:11.574753119 -0400
***************
*** 1,36 ****
! ;;; wisent-python-wy.el --- Generated parser support file
  
! ;; Copyright (C) 2002-2011 Free Software Foundation, Inc.
  
! ;; Author: Chong Yidong <cyd@furball>
! ;; Created: 2011-07-31 20:19:50-0400
! ;; Keywords: syntax
! ;; X-RCS: $Id$
  
! ;; This file is not part of GNU Emacs.
  
! ;; This program is free software; you can redistribute it and/or
! ;; modify it under the terms of the GNU General Public License as
! ;; published by the Free Software Foundation, either version 3 of
! ;; the License, or (at your option) any later version.
! 
! ;; This software is distributed in the hope that it will be useful,
  ;; but WITHOUT ANY WARRANTY; without even the implied warranty of
! ;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
! ;; General Public License for more details.
! ;;
  ;; You should have received a copy of the GNU General Public License
! ;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
  
  ;;; Commentary:
  ;;
! ;; PLEASE DO NOT MANUALLY EDIT THIS FILE!  It is automatically
! ;; generated from the grammar file python.wy.
! 
! ;;; History:
  ;;
  
  ;;; Code:
  
  ;;; Prologue
  ;;
--- 1,81 ----
! ;;; semantic/wisent/python-wy.el --- Generated parser support file
  
! ;; Copyright (C) 2002-2004, 2007, 2010-2011  Free Software Foundation, Inc.
! ;; Copyright (C) 2001-2010 Python Software Foundation
  
! ;; This file is part of GNU Emacs.
  
! ;; GNU Emacs is free software: you can redistribute it and/or modify
! ;; it under the terms of the GNU General Public License as published by
! ;; the Free Software Foundation, either version 3 of the License, or
! ;; (at your option) any later version.
  
! ;; GNU Emacs is distributed in the hope that it will be useful,
  ;; but WITHOUT ANY WARRANTY; without even the implied warranty of
! ;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
! ;; GNU General Public License for more details.
! 
  ;; You should have received a copy of the GNU General Public License
! ;; along with GNU Emacs.  If not, see <http://www.gnu.org/licenses/>.
  
  ;;; Commentary:
  ;;
! ;; This file was generated from etc/grammars/python.wy.
! ;; It is derived in part from the Python grammar, used under the
! ;; following license:
! ;;
! ;; PYTHON SOFTWARE FOUNDATION LICENSE VERSION 2
! ;; --------------------------------------------
! ;; 1. This LICENSE AGREEMENT is between the Python Software Foundation
! ;; ("PSF"), and the Individual or Organization ("Licensee") accessing
! ;; and otherwise using this software ("Python") in source or binary
! ;; form and its associated documentation.
! ;;
! ;; 2. Subject to the terms and conditions of this License Agreement,
! ;; PSF hereby grants Licensee a nonexclusive, royalty-free, world-wide
! ;; license to reproduce, analyze, test, perform and/or display
! ;; publicly, prepare derivative works, distribute, and otherwise use
! ;; Python alone or in any derivative version, provided, however, that
! ;; PSF's License Agreement and PSF's notice of copyright, i.e.,
! ;; "Copyright (c) 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008,
! ;; 2009, 2010 Python Software Foundation; All Rights Reserved" are
! ;; retained in Python alone or in any derivative version prepared by
! ;; Licensee.
! ;;
! ;; 3. In the event Licensee prepares a derivative work that is based
! ;; on or incorporates Python or any part thereof, and wants to make
! ;; the derivative work available to others as provided herein, then
! ;; Licensee hereby agrees to include in any such work a brief summary
! ;; of the changes made to Python.
! ;;
! ;; 4. PSF is making Python available to Licensee on an "AS IS"
! ;; basis.  PSF MAKES NO REPRESENTATIONS OR WARRANTIES, EXPRESS OR
! ;; IMPLIED.  BY WAY OF EXAMPLE, BUT NOT LIMITATION, PSF MAKES NO AND
! ;; DISCLAIMS ANY REPRESENTATION OR WARRANTY OF MERCHANTABILITY OR FITNESS
! ;; FOR ANY PARTICULAR PURPOSE OR THAT THE USE OF PYTHON WILL NOT
! ;; INFRINGE ANY THIRD PARTY RIGHTS.
  ;;
+ ;; 5. PSF SHALL NOT BE LIABLE TO LICENSEE OR ANY OTHER USERS OF PYTHON
+ ;; FOR ANY INCIDENTAL, SPECIAL, OR CONSEQUENTIAL DAMAGES OR LOSS AS A
+ ;; RESULT OF MODIFYING, DISTRIBUTING, OR OTHERWISE USING PYTHON, OR
+ ;; ANY DERIVATIVE THEREOF, EVEN IF ADVISED OF THE POSSIBILITY THEREOF.
+ ;;
+ ;; 6. This License Agreement will automatically terminate upon a
+ ;; material breach of its terms and conditions.
+ ;;
+ ;; 7. Nothing in this License Agreement shall be deemed to create any
+ ;; relationship of agency, partnership, or joint venture between PSF
+ ;; and Licensee.  This License Agreement does not grant permission to
+ ;; use PSF trademarks or trade name in a trademark sense to endorse or
+ ;; promote products or services of Licensee, or any third party.
+ ;;
+ ;; 8. By copying, installing or otherwise using Python, Licensee
+ ;; agrees to be bound by the terms and conditions of this License
+ ;; Agreement.
  
  ;;; Code:
+ 
+ (require 'semantic/lex)
  
  ;;; Prologue
  ;;
***************
*** 40,46 ****
  (defconst wisent-python-wy--keyword-table
    (semantic-lex-make-keyword-table
     '(("and" . AND)
-      ("as" . AS)
       ("assert" . ASSERT)
       ("break" . BREAK)
       ("class" . CLASS)
--- 85,90 ----
***************
*** 78,84 ****
       ("pass" summary "Statement that does nothing")
       ("or" summary "Binary logical 'or' operator")
       ("not" summary "Unary boolean negation operator")
-      ("lambda" summary "Create anonymous function")
       ("is" summary "Binary operator that tests for object equality")
       ("in" summary "Part of 'for' statement ")
       ("import" summary "Load specified modules")
--- 122,127 ----
***************
*** 93,103 ****
       ("elif" summary "Shorthand for 'else if' following an 'if' statement")
       ("del" summary "Delete specified objects, i.e., undo what assignment did")
       ("def" summary "Define a new function")
!      ("continue" summary "Skip to the next iteration of enclosing 'for' or 'while' loop")
       ("class" summary "Define a new class")
       ("break" summary "Terminate 'for' or 'while' loop")
       ("assert" summary "Raise AssertionError exception if <expr> is false")
-      ("as" summary "EXPR as NAME makes value of EXPR available as variable NAME")
       ("and" summary "Logical AND binary operator ... ")))
    "Table of language keywords.")
  
--- 136,145 ----
       ("elif" summary "Shorthand for 'else if' following an 'if' statement")
       ("del" summary "Delete specified objects, i.e., undo what assignment did")
       ("def" summary "Define a new function")
!      ("continue" summary "Skip to the next interation of enclosing 'for' or 'while' loop")
       ("class" summary "Define a new class")
       ("break" summary "Terminate 'for' or 'while' loop")
       ("assert" summary "Raise AssertionError exception if <expr> is false")
       ("and" summary "Logical AND binary operator ... ")))
    "Table of language keywords.")
  
***************
*** 180,186 ****
      (eval-when-compile
        (require 'semantic/wisent/comp))
      (wisent-compile-grammar
!      '((BACKSLASH NEWLINE INDENT DEDENT INDENT_BLOCK PAREN_BLOCK BRACE_BLOCK BRACK_BLOCK LPAREN RPAREN LBRACE RBRACE LBRACK RBRACK LTLTEQ GTGTEQ EXPEQ DIVDIVEQ DIVDIV LTLT GTGT EXPONENT EQ GE LE PLUSEQ MINUSEQ MULTEQ DIVEQ MODEQ AMPEQ OREQ HATEQ LTGT NE HAT LT GT AMP MULT DIV MOD PLUS MINUS PERIOD TILDE BAR COLON SEMICOLON COMMA ASSIGN BACKQUOTE STRING_LITERAL NUMBER_LITERAL NAME AND AS ASSERT BREAK CLASS CONTINUE DEF DEL ELIF ELSE EXCEPT EXEC FINALLY FOR FROM GLOBAL IF IMPORT IN IS LAMBDA NOT OR PASS PRINT RAISE RETURN TRY WHILE YIELD)
         nil
         (goal
  	((NEWLINE))
--- 222,228 ----
      (eval-when-compile
        (require 'semantic/wisent/comp))
      (wisent-compile-grammar
!      '((BACKSLASH NEWLINE INDENT DEDENT INDENT_BLOCK PAREN_BLOCK BRACE_BLOCK BRACK_BLOCK LPAREN RPAREN LBRACE RBRACE LBRACK RBRACK LTLTEQ GTGTEQ EXPEQ DIVDIVEQ DIVDIV LTLT GTGT EXPONENT EQ GE LE PLUSEQ MINUSEQ MULTEQ DIVEQ MODEQ AMPEQ OREQ HATEQ LTGT NE HAT LT GT AMP MULT DIV MOD PLUS MINUS PERIOD TILDE BAR COLON SEMICOLON COMMA ASSIGN BACKQUOTE STRING_LITERAL NUMBER_LITERAL NAME AND ASSERT BREAK CLASS CONTINUE DEF DEL ELIF ELSE EXCEPT EXEC FINALLY FOR FROM GLOBAL IF IMPORT IN IS LAMBDA NOT OR PASS PRINT RAISE RETURN TRY WHILE YIELD)
         nil
         (goal
  	((NEWLINE))
***************
*** 288,296 ****
  	((testlist)
  	 nil))
         (yield_stmt
- 	((YIELD)
- 	 (wisent-raw-tag
- 	  (semantic-tag-new-code $1 nil)))
  	((YIELD testlist)
  	 (wisent-raw-tag
  	  (semantic-tag-new-code $1 nil))))
--- 330,335 ----
***************
*** 331,344 ****
  	((import_as_name_list COMMA import_as_name)
  	 nil))
         (import_as_name
! 	((NAME as_name_opt)
  	 nil))
         (dotted_as_name
! 	((dotted_name as_name_opt)))
!        (as_name_opt
  	(nil)
! 	((AS NAME)
! 	 (identity $2)))
         (dotted_name
  	((NAME))
  	((dotted_name PERIOD NAME)
--- 370,383 ----
  	((import_as_name_list COMMA import_as_name)
  	 nil))
         (import_as_name
! 	((NAME name_name_opt)
  	 nil))
         (dotted_as_name
! 	((dotted_name name_name_opt)))
!        (name_name_opt
  	(nil)
! 	((NAME NAME)
! 	 nil))
         (dotted_name
  	((NAME))
  	((dotted_name PERIOD NAME)
***************
*** 670,676 ****
     '((parse-stream . wisent-parse-stream)))
    (setq semantic-parser-name "LALR"
  	semantic--parse-table wisent-python-wy--parse-table
! 	semantic-debug-parser-source "python.wy"
  	semantic-flex-keywords-obarray wisent-python-wy--keyword-table
  	semantic-lex-types-obarray wisent-python-wy--token-table)
    ;; Collect unmatched syntax lexical tokens
--- 709,715 ----
     '((parse-stream . wisent-parse-stream)))
    (setq semantic-parser-name "LALR"
  	semantic--parse-table wisent-python-wy--parse-table
! 	semantic-debug-parser-source "wisent-python.wy"
  	semantic-flex-keywords-obarray wisent-python-wy--keyword-table
  	semantic-lex-types-obarray wisent-python-wy--token-table)
    ;; Collect unmatched syntax lexical tokens
***************
*** 680,687 ****
  
  
  ;;; Analyzers
- ;;
- (require 'semantic-lex)
  
  (define-lex-keyword-type-analyzer wisent-python-wy--<keyword>-keyword-analyzer
    "keyword analyzer for <keyword> tokens."
--- 719,724 ----
***************
*** 757,762 ****
  ;;; Epilogue
  ;;
  
! (provide 'wisent-python-wy)
  
! ;;; wisent-python-wy.el ends here
--- 794,799 ----
  ;;; Epilogue
  ;;
  
! (provide 'semantic/wisent/python-wy)
  
! ;;; semantic/wisent/python-wy.el ends here
